scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1200,6 +1225,56 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 	return false;
 }
 
+/*
+ * Helper function to consume layer DSQs when a system is under starvation.
+ */
+static __always_inline bool consume_starved(s32 cpu)
+{
+	struct layer *layer;
+	u32 idx;
+	u64 decay_vtime_sum = 0;
+	u64 layer_weight_sum = 0;
+	u64 decay_proportion_vtime = 0;
+	u64 vtime_weighted_decay = 0;
+
+	bpf_for(idx, 0, nr_layers) {
+		layer = MEMBER_VPTR(layers, [idx]);
+		decay_vtime_sum += layer->vtime_decay;
+		layer_weight_sum += layer->weight;
```

Comment:
We did mention extracting this to the Rust, but it's not a big deal to do it here. Adding for reference.